### PR TITLE
[Doc]--Introduce GenericRecordBuilder and the Avro implementation

### PR DESCRIPTION
**Pull request by Huanli-Meng**
### Motivation

Release 2.4.0 introduces the generic record builder and provides a AVRO generic record implementation. here is the PR link: https://github.com/apache/pulsar/pull/3690. But the document is not updated accordingly.

Therefore, the PR is created to updated the document. docs in the following releases are updated:
releases 2.4.0/2.4.1/2.4.2/2.5.0

Docs in releases 2.5.1 and later releases have been re-structured and updated.

### Modifications

Add an example in Schema registry to describe how to use the GenericRecordBuilder to define the schema and access data.

### CODE REVIEW

**Comment by Huanli-Meng** [comment]
@jiazhai , please help review the PR. Thanks.

File: site2/website/versioned_docs/version-2.4.0/concepts-schema-registry.md [hunk 1]
```diff
@@ -79,6 +79,28 @@ For usage instructions, see the documentation for your preferred client library:
 
 > Support for other schema formats will be added in future releases of Pulsar.
 
+The following example shows how to define a Avro schema using the `GenericSchemaBuilder`, generate a generic Avro schema using `GenericRecordBuilder`, and consume messages into `GenericRecord`.
```
Comment (Anonymitaet) [inline site2/website/versioned_docs/version-2.4.0/concepts-schema-registry.md:82]:
```suggestion
The following example shows how to define an Avro schema using the `GenericSchemaBuilder`, generate a generic Avro schema using `GenericRecordBuilder`, and consume messages into `GenericRecord`.
```

pls check all

**Comment by Huanli-Meng** [comment]
will check all docs and fix them one by one. Thx

**Comment by Huanli-Meng** [comment]
@Anonymitaet , docs have been updated, please help double check. Thanks.

**Pull request merged**